tpdatasrc/tpgamefiles/rules/char_class/class012_monk.py

```diff
@@ -40,72 +40,81 @@ def GetClassHelpTopic():
-	
+    return base_attack_bonus_type_semi_martial
+    
 def IsFortSaveFavored():
-	return 1
-	
+    return 1
+    
 def IsRefSaveFavored():
-	return 1
-	
+    return 1
+    
 def IsWillSaveFavored():
-	return 1
+    return 1
 
 def GetSpellListType():
-	return spell_list_type_none
+    return spell_list_type_none
 
 def IsClassSkill(skillEnum):
-	return char_class_utils.IsClassSkill(class_skills, skillEnum)
+    return char_class_utils.IsClassSkill(class_skills, skillEnum)
 
 def IsClassFeat(featEnum):
-	return char_class_utils.IsClassFeat(class_feats, featEnum)
+    return char_class_utils.IsClassFeat(class_feats, featEnum)
 
 def GetClassFeats():
-	return class_feats
+    return class_feats
 
 def IsAlignmentCompatible( alignment):
-	if (alignment & ALIGNMENT_LAWFUL) != 0:
-		return 1
-	return 0
+    if (alignment & ALIGNMENT_LAWFUL) != 0:
+        return 1
+    return 0
 
 def ObjMeetsPrereqs( obj ):
-	char_classes = obj.char_classes
-	if char_classes.count > 0:
-		monkLevel = obj.stat_level_get(stat_level_monk)
-		if monkLevel > 0 and char_classes[len(char_classes) - 1] != stat_level_monk:
-			return 0
-	return 1
+    char_classes = obj.char_classes
+    if char_classes.count > 0:
+        monkLevel = obj.stat_level_get(stat_level_monk)
+        if monkLevel > 0 and char_classes[len(char_classes) - 1] != stat_level_monk:
+            return 0
+    return 1
 
 # Levelup
 
 def IsSelectingFeatsOnLevelup( obj ):
-	newLvl = char_editor.stat_level_get( classEnum )
-	if newLvl == 2:
-		if obj.has_feat(feat_combat_reflexes) and obj.has_feat(feat_deflect_arrows):
-			return 0
-		return 1
-	if newLvl == 6:
-		if obj.has_feat(feat_improved_trip) and obj.has_feat(feat_improved_disarm):
-			return 0
-		return 1
-	return 0
+    newLvl = char_editor.stat_level_get( classEnum )
+    if newLvl == 2:
+        if obj.has_feat(feat_combat_reflexes) and obj.has_feat(feat_deflect_arrows) and obj.has_feat("Fiery Fist"):
+            return 0
+        return 1
+    if newLvl == 6:
+        if obj.has_feat(feat_improved_trip) and obj.has_feat(feat_improved_disarm):
+            return 0
+        return 1
+    return 0
 
 def LevelupGetBonusFeats(obj):
-	newLvl = char_editor.stat_level_get( classEnum )
-	bonus_feats = []
-	if newLvl == 2:
-		bonus_feats = [feat_combat_reflexes, feat_deflect_arrows]
-	if newLvl == 6:
-		bonus_feats = [feat_improved_trip, feat_improved_disarm]
-	
-	bonFeatInfo = []
-	for ft in bonus_feats:
-		bonFeatInfo.append(char_editor.FeatInfo(ft))
-	char_editor.set_bonus_feats(bonFeatInfo)
-	return
\ No newline at end of file
+    newLvl = char_editor.stat_level_get( classEnum )
+    bonus_feats = []
+    if newLvl == 2:
+        bonus_feats = [feat_combat_reflexes, feat_deflect_arrows]
+        #I know, atm a monk has to have stunning fist
+        #but maybe improved grapple will be added as a choice at first level
+        if char_editor.has_feat(feat_stunning_fist):
+            bonus_feats.append("Fiery Fist")
```

Comment:
Hmm, this is new content, does it show up if new content isn't enabled?

Reply:
No it doesn't show up, the feats are all flagged non core and I've tested it with Non Core options disabled, when you level up a monk you can't see the feat and therefor you can't select it as a bonus feat as well.